pkg/streamingpromql/types/limiting_pool.go

```diff
@@ -223,6 +223,11 @@ func (p *LimitingBucketedPool[S, E]) Put(s S, tracker *limiter.MemoryConsumption
 	}
 
 	tracker.DecreaseMemoryConsumption(uint64(cap(s))*p.elementSize, p.source)
+	if seriesMetadataSlice, isSeriesMetadataSlice := any(s).([]SeriesMetadata); isSeriesMetadataSlice {
+		for _, sm := range seriesMetadataSlice {
+			tracker.DecreaseMemoryConsumptionForLabels(sm.Labels)
+		}
+	}
```

Comment:
Rather than adding this type check here, I would add an extra method as a field on `LimitingBucketedPool` that is called by `Put`. Then, when we create the `LimitingBucketedPool` for `SeriesMetadata` slices, we can pass a method that adjusts the memory consumption estimate based on the labels in the slice before we put the slice in the pool.